File: Rotate List/Rotate List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode rotateRight(ListNode head, int n) {
        if(n == 0){
            return head;
        }
        if(head == null){
            return head;
        }
        
        ListNode slowPtr = head;
        ListNode fastPtr = head;
        
        while(n-- > 0){
            if(fastPtr.next != null){
                fastPtr = fastPtr.next;
            }
            else{
                fastPtr = head;
            }
        }
        
        if(slowPtr == fastPtr){
            return head;
        }
        
        while(fastPtr.next != null){
            slowPtr = slowPtr.next;
            fastPtr = fastPtr.next;
        }
        
        ListNode retval = slowPtr.next;
        fastPtr.next = head;
        slowPtr.next = null;
        
        return retval;
    }
}

File: Rotate List/Rotate List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param head, a ListNode
    # @param k, an integer
    # @return a ListNode
    def rotateRight(self, head, k):
        
        if head == None:
            return None
        
        temp = head
        for i in xrange(0, k):
            if temp.next == None:
                temp = head
            else:
                temp = temp.next
                
        newLast = head
        while temp.next != None:
            temp = temp.next
            newLast = newLast.next
            
        temp.next = head
        newHead = newLast.next
        newLast.next = None
        return newHead
